verilog/tools/ls/autoexpand.cc

```diff
@@ -204,32 +312,38 @@ const std::regex AutoExpander::autotemplate_type_re_{
 const std::regex AutoExpander::autotemplate_conn_re_{
     R"(\.\s*([^\s(]+?)\s*\(\s*([^\s(]+?)\s*\))"};
 
-const AutoExpander::Template *AutoExpander::Template::from(
-    const Map &templates, const absl::string_view type_id,
-    const int64_t instance_offset) {
-  const auto templates_it = templates.find(type_id);
-  const Template *matching_tmpl = nullptr;
-  if (templates_it != templates.end()) {
-    // Linear search for the matching template (there should be very few
-    // templates per type, often just one)
-    for (const auto &tmpl : templates_it->second) {
-      if (instance_offset < tmpl.offset) break;
-      matching_tmpl = &tmpl;
-    }
-  }
-  return matching_tmpl;
-}
-
-void AutoExpander::Module::EmitPortDeclarations(
+// AUTOINPUT/OUTPUT/INOUT/WIRE/REG regex breakdown:
+// (/\*\s* ... \s*\*/\s*?\n)                – starting comment
+// (?:\s*//.*\n)?                           – optional starting comment
+//                                            ("Beginning of automatic...")
+// (?: ... )?                               – an optional non-capturing group:
+//   ([\s\S]*?)?                            – any text (usually port
+//                                            declarations)
+//   (?:[^\S\r\n]*// End of automatics.*\n) – ended by an "End of automatics"
+//                                            comment
+constexpr const char *auto_declaration_re_fmt =
```

Comment:
that can be a `static constexpr absl::string_view` - that way, the length is already compile-time initialized and doesn't have to be determined with a `strlen()` at runtime.